### ISSUE TITLE
UnInstall-BcContainerApp -doNotSaveData requires values

### ISSUE THREAD

**Issue by mhenricus** [microsoft/navcontainerhelper#2245]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
I'd like to completely remove an app from the sandbox with the UnInstall-BcContainerApp command, so including the data and schema. I used command below, but then I have to supply values for parameters for Name: 
But what to enter here? I tried the app name (though it is already given as parameter), but it always fails.

**Scripts used to create container and cause the issue**
```
UnInstall-BcContainerApp -containerName $containerName -appName $appName -doNotSaveData -doNotSaveSchema
```

**Full output of scripts**
```
Uninstalling <App> from default
Cleaning Schema from <App> on default
cmdlet Sync-NAVApp at command pipeline position 1
Supply values for the following parameters:
Name:
...


Error:
ServerInstance 'MicrosoftDynamicsNavServer$System.Collections.Hashtable' contains at least one invalid character. You cannot use the following characters: '.'at <ScriptBlock>, <No file>: line 21
at Invoke-ScriptInBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.16\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 44
at UnInstall-BcContainerApp, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.16\AppHandling\UnInstall-NavContainerApp.ps1: line 47
at <ScriptBlock>, <No file>: line 1
ServerInstance 'MicrosoftDynamicsNavServer$System.Collections.Hashtable' contains at least one invalid character. You cannot use the following characters: '.
'
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.16\ContainerHandling\Invoke-ScriptInNavContainer.ps1:45 char:13
+             throw $_.Exception.Message
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (ServerInstance ...characters: '.':String) [], RuntimeException
    + FullyQualifiedErrorId : ServerInstance 'MicrosoftDynamicsNavServer$System.Collections.Hashtable' contains at least one invalid character. You cannot u 
   se the following characters: '.'

```

**Screenshots**

**Additional context**
- does it happen all the time?
- did it use to work?

**Comment by freddydk** [comment]
Please try with the latest version of BcContainerHelper.
If that still fails, please indicate which version of Business Central this happens in?

Note that in order to totally remove an app - you probably need to use unpublish-bccontainerapp with -uninstall -donotsavedata and -donotsaveschema

**Comment by mhenricus** [comment]
Hi Freddy, 

UnPublish-BcContainerApp does the trick. Thanks!
UnPublish-BcContainerApp -containerName $containerName -appName $appName -version $newVersion -unInstall -doNotSaveData -doNotSaveSchema -force


But, then why does UnInstall-BcContainerApp have the parameters -doNotSaveData -doNotSaveSchema? I guess that misleaded me.

To reproduce the error with the UnInstall-BcContainerApp:
I installed the app on a clean container with the following commands:

$newVersion = '0.0.0.0'
$newVersionFile = 'C:\Users\maart\Documents\AL\ISM\app\IgoDynamics_Subscription Management_' + $newVersion + '.app'
$containerName = 'l19nl-1215'
$appName = 'Subscription Management'
Publish-BcContainerApp -containerName $containerName -appFile $newVersionFile -skipVerification
Sync-BcContainerApp -containerName $containerName -appName $appName -appVersion $newVersion
Install-BcContainerApp -containerName $containerName -appName $appName -appVersion $newVersion

For uninstalling I used the following command and get the error.

PS D:\OneDrive - Intelligo\BC\PowerShell> UnInstall-BcContainerApp -containerName $containerName -name $appName -doNotSaveData -doNotSaveSchema
Uninstalling Subscription Management from default
Cleaning Schema from Subscription Management on default

cmdlet Sync-NAVApp at command pipeline position 1
Supply values for the following parameters:
Name: Subscription Management
ServerInstance 'MicrosoftDynamicsNavServer$System.Collections.Hashtable' contains at least one invalid character. You cannot use the following characters: '.'
at <ScriptBlock>, <No file>: line 21
at Invoke-ScriptInBcContainer, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.0\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 43
at UnInstall-BcContainerApp, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.0\AppHandling\UnInstall-NavContainerApp.ps1: line 47
at <ScriptBlock>, <No file>: line 1
UnInstall-BcContainerApp Telemetry Correlation Id: 3588e470-4d90-4d0d-be73-48b2b56e9ddd
ServerInstance 'MicrosoftDynamicsNavServer$System.Collections.Hashtable' contains at least one invalid character. You cannot use the following characters: '.'
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.0\ContainerHandling\Invoke-ScriptInNavContainer.ps1:44 char:13
+             throw $_.Exception.Message
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (ServerInstance ...characters: '.':String) [], RuntimeException
    + FullyQualifiedErrorId : ServerInstance 'MicrosoftDynamicsNavServer$System.Collections.Hashtable' contains at least one invalid character. You cannot use the following characters: '.'

**Comment by freddydk** [comment]
Found the reason why the uninstall function didn't work
Will be fixed in the next version

**Comment by freddydk** [comment]
Fixed in 3.0.1
